packages/manager/src/utilities/sort-by.test.ts

```diff
@@ -0,0 +1,218 @@
+import {
+  sortByString,
+  sortByUTFDate,
+  sortByNumber,
+  sortByArrayLength,
+  sortByRegionLabel,
+  toComparator,
+} from './sort-by';
+
+describe('sortByString', () => {
+  it('returns -1 when the first value comes first alphabetically', () => {
+    expect(sortByString('hello', 'HELLO')).toBe(-1);
+    expect(sortByString('hello', 'world')).toBe(-1);
+    expect(sortByString('hello', 'world', 'asc')).toBe(-1);
+    expect(sortByString('Hello', 'Hello!')).toBe(-1);
+    expect(sortByString('Linode 1', 'Linode 2')).toBe(-1);
+    expect(sortByString('Linode 1', 'Linode 10')).toBe(-1);
+  });
+
+  it('returns 1 when the second value comes first alphabetically', () => {
+    expect(sortByString('HELLO', 'hello')).toBe(1);
+    expect(sortByString('world', 'hello')).toBe(1);
+    expect(sortByString('world', 'hello', 'asc')).toBe(1);
+    expect(sortByString('Hello!', 'Hello')).toBe(1);
+    expect(sortByString('Linode 2', 'Linode 1')).toBe(1);
+    expect(sortByString('Linode 10', 'Linode 1')).toBe(1);
+  });
+
+  it('returns 0 when the values are equal', () => {
+    expect(sortByString('hello', 'hello')).toBe(0);
+    expect(sortByString('hello', 'hello', 'asc')).toBe(0);
+    expect(sortByString('hello', 'hello', 'desc')).toBe(0);
+  });
+
+  it('returns inverted values when sort order is descending', () => {
+    expect(sortByString('hello', 'world', 'desc')).toBe(1);
+    expect(sortByString('world', 'hello', 'desc')).toBe(-1);
+    expect(sortByString('hello', 'hello', 'desc')).toBe(0);
+  });
```

Comment:
feedback: Can we remove some redundant tests with below snippet ?

```suggestion
it('Should sort alphabetically default in asc order', () => {
    const actual = ['cherry', 'banana', 'apple', 'Linode 2', 'Linode 1'];
    const expected = ['apple', 'banana', 'cherry', 'Linode 1', 'Linode 2'];
    expect(actual.sort(sortByString)).toStrictEqual(expected);
  });
  it('Should sort alphabetically in desc order', () => {
    const actual = ['apple', 'banana', 'cherry', 'Linode 1', 'Linode 2'];
    const expected = ['Linode 2', 'Linode 1', 'cherry', 'banana', 'apple'];
    expect(actual.sort((a, b) => sortByString(a, b, 'desc'))).toStrictEqual(
      expected
    );
  });
```